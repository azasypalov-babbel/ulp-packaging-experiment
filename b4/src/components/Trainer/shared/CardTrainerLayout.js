import React from 'react';
import PropTypes from 'prop-types';

import { rem } from 'polished';
import styled from 'styled-components';

import * as features from '../../../lib/features';

import TrainerTitle from './TrainerTitle';
import TrainerImage from './TrainerImage';

const StyledWrapper = styled.div`
  display: flex;
  flex-direction: column;

  padding: 0 1rem;
  margin: 0 auto;

  max-width: ${rem(736)};
  width: 100%;

  box-sizing: border-box;

  ${() => {
    if (features.isWebview()) {
      return 'user-select: none;';
    }
  }};
`;

const StyledContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;

  @media (min-width: ${(props) => props.theme.viewports.breakpoints.xsmall}) {
    flex-direction: row;
  }

  & p {
    margin-bottom: 1rem;
  }
`;

const CardTrainerLayout = ({
  titleText,
  children,
  imageId,
  ...props
}) => (
  <StyledWrapper {...props}>
    <TrainerTitle text={titleText} />
    <StyledContainer>
      <div>
        {children}
      </div>
      <TrainerImage imageId={imageId} />
    </StyledContainer>
  </StyledWrapper>
);

CardTrainerLayout.propTypes = {
  titleText: PropTypes.string.isRequired,
  children: PropTypes.node,
  imageId: PropTypes.string
};

export default CardTrainerLayout;

