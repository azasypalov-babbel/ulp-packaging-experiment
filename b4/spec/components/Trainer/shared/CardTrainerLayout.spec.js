import React from 'react';
import { shallow } from 'enzyme';

import CardTrainerLayout from '../../../../src/components/Trainer/shared/CardTrainerLayout';

const defaultProps = {
  dataSelector: 'card-data-selector',
  titleText: 'card title',
  imageId: 'image-id'
};

describe('<CardTrainerLayout />', () => {
  test('it renders', () => {
    const component = shallow(
      <CardTrainerLayout {...defaultProps}>
        <h1>A card row</h1>
      </CardTrainerLayout>
    );

    expect(component).toMatchSnapshot();
  });
});
