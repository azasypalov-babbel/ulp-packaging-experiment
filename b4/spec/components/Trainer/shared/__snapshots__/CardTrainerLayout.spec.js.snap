// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardTrainerLayout /> it renders 1`] = `
<styled.div
  dataSelector="card-data-selector"
>
  <TrainerTitle
    text="card title"
  />
  <styled.div>
    <div>
      <h1>
        A card row
      </h1>
    </div>
    <WithServices(TrainerImage)
      imageId="image-id"
    />
  </styled.div>
</styled.div>
`;
